src/js/core/UpdateStream.ts

```diff
@@ -0,0 +1,145 @@
+import { Observable } from "rxjs/Observable";
+import compareVersions from "compare-versions";
+import { request, RequestResponse } from "@dcos/http-service";
+import { BehaviorSubject } from "rxjs";
+
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+
+export const UpdateStreamType: symbol = Symbol("UpdateStreamType");
+const CHECK_DELAY: number = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION_SETTING: string = "dcosUIVersion";
+const DISMISSED_VERSION: string = "dismissedVersion";
+const LAST_TIME_CHECK: string = "lastTimeCheck";
+
+interface ContentTypeObject {
+  action: string;
+  actionType: string;
+  entity: string;
+  version: string;
+}
+
+interface DCOSUserSettings {
+  dcosUIVersion: DCOSUIVersion;
+  [key: string]: DCOSUIVersion;
+}
+
+interface DCOSUIVersion {
+  dismissedVersion: string;
+  lastTimeCheck: number;
+  [key: string]: string | number;
+}
+
+export function showNotification(newVersion: string): void {
+  // TODO: Move inside UpdateServiceBanner
+  alert(newVersion);
+}
+
+function getVersionFromVersionObject(versionObject: any): string {
+  return Object.keys(versionObject.response.results)[0];
+}
+
+function getFromLocalStorage(key: string): any {
+  const value: any = Util.findNestedPropertyInObject(
+    UserSettingsStore.getKey(SAVED_STATE_KEY),
+    DCOS_UI_VERSION_SETTING + "." + key
+  );
+
+  return value;
+}
+
+function setInLocalStorage(key: string, value: any): void {
+  const savedStates: DCOSUserSettings =
+    UserSettingsStore.getKey(SAVED_STATE_KEY) || {};
+  savedStates[DCOS_UI_VERSION_SETTING] = {
+    dismissedVersion: localStorageDismissedVersion.getValue(),
+    lastTimeCheck: localStorageCheckedTime.getValue()
+  };
+  savedStates[DCOS_UI_VERSION_SETTING][key] = value;
+  UserSettingsStore.setKey(SAVED_STATE_KEY, savedStates);
+  if (key === DISMISSED_VERSION) {
+    localStorageDismissedVersion.next(value);
+  } else if (key === LAST_TIME_CHECK) {
+    localStorageCheckedTime.next(value);
+  }
+}
+
+function getContentType({
+  action,
+  actionType,
+  entity,
+  version
+}: ContentTypeObject): string {
+  return `application/vnd.dcos.${entity}.${action}-${actionType}+json;charset=utf-8;version=${version}`;
+}
+
+const localStorageDismissedVersion: BehaviorSubject<
+  string
+> = new BehaviorSubject(
+  getFromLocalStorage(DISMISSED_VERSION)
+    ? getFromLocalStorage(DISMISSED_VERSION)
+    : "0"
```

Comment:
Shouldn't this be sth like null instead? A string zero seems to be strange to identify as much as "nothing" (from my understanding)